src/models/queues.ts

```diff
@@ -84,6 +84,10 @@ export interface Queue {
      * The Entries of the Queue
      */
     entries: QueueEntry[],
+    /**
+     * The callback functions for all subscribers
+     */
+    callbackFunctions: Map<string, () => void>
```

Comment:
Das hier muss noch etwas überarbeitet werden:
- Die Lambda-Funktionen sollten auch asyncron sein dürfen. Kann man hier trotzdem über void machen aber sollte man dann anmerken dass nicht auf die Callback-Funktionen gewartet wird
- Es ist keine gute Idee, Lambda-Funktionen in der Datenbank zu speichern, weil das zu ACE führen kann
- Stattdessen sollten wir eine eigene Datenklasse dafür anlegen, z.B. queueSubscriptions die nur eine Beziehung von Nutzer auf Queue speichert (könnte man dann auch ggf. noch zukünftig typsicher erweitern, z.B. durch verschiedene Arten von Subscriptions: all_join, relevant_join(also nicht wenn man gerade in einem Sprechstundenkanal ist), talk_timer (z.B. nach 15 Minuten),...)
- Imo würde das auch besser in die UserSessions passen als das in die queues mit rein zu packen
- Du hast vergessen das im Schema anzupassen, also würde es für die Datenbank keinen Effekt haben. Da müsste noch z.B. sowas rein:
```ts
callbackFunctions: {
        type: mongoose.SchemaTypes.Map,
        required: false,
        default: new Map<string, () => void>(),
    },
```
(wobei das so nicht gehen würde, weil man keine Functions direkt speichern kann... Man müsste die Function zu nem String konvertieren was wie oben erwähnt zu ACE exploits führen wird)